executor/index_merge_reader_test.go

```diff
@@ -273,23 +282,67 @@ func (s *testSuite1) TestIndexMergeInTransaction(c *C) {
 	tk.MustExec("insert into t1 values(11, 11, 11, 11, 11);")
 	tk.MustExec("insert into t1 values(21, 21, 21, 21, 21);")
 	tk.MustExec("insert into t1 values(31, 31, 31, 31, 31);")
-	res := tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < 20) and c3 < 20;").Sort()
+
+	res := tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < -1 or c2 < 20) and c3 < 20;").Sort()
+	res.Check(testkit.Rows("1 1 1 1 1", "11 11 11 11 11"))
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < -1) and c3 < 20;").Sort()
+	res.Check(testkit.Rows("1 1 1 1 1", "11 11 11 11 11"))
+
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < -1 or c2 < 20) and c3 < 20;").Sort()
 	res.Check(testkit.Rows("1 1 1 1 1", "11 11 11 11 11"))
-	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < 20) and c3 < 20;").Sort()
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < -1) and c3 < 20;").Sort()
 	res.Check(testkit.Rows("1 1 1 1 1", "11 11 11 11 11"))
 
 	tk.MustExec("update t1 set c3 = 100 where c3 = 1;")
-	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < 20) and c3 < 20;")
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < -1 or c2 < 20) and c3 < 20;")
 	res.Check(testkit.Rows("11 11 11 11 11"))
-	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < 20) and c3 < 20;")
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < -1) and c3 < 20;")
+	res.Check(testkit.Rows("11 11 11 11 11"))
+
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < -1 or c2 < 20) and c3 < 20;")
+	res.Check(testkit.Rows("11 11 11 11 11"))
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < -1) and c3 < 20;")
 	res.Check(testkit.Rows("11 11 11 11 11"))
 
 	tk.MustExec("delete from t1;")
-	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < 20) and c3 < 20;")
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < -1 or c2 < 20) and c3 < 20;")
+	res.Check(testkit.Rows())
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (c1 < 20 or c2 < -1) and c3 < 20;")
 	res.Check(testkit.Rows())
-	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < 20) and c3 < 20;")
+
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < -1 or c2 < 20) and c3 < 20;")
+	res.Check(testkit.Rows())
+	res = tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where (pk < 20 or c2 < -1) and c3 < 20;")
 	res.Check(testkit.Rows())
 	tk.MustExec("commit;")
+
+	// This is a case generated by sqlgen to test if clusted index is ok.
```

Comment:
```suggestion
	// This is a case generated by sqlgen to test if clustered index is ok.
```

Reply:
Maybe we could add some compound primary clustered index cases with union scan executor.

Reply:
OK. I will add more case to test:
1. Compoud primary clustered key(int col + other col)
2. Single primary key, but is string or other type.